source/framework/vx_copy_swap.c

```diff
@@ -0,0 +1,571 @@
+/*
+
+ * Copyright (c) 2012-2017 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/*
+Implementation file for the copy and swap utilities
+*/
+
+#include <vx_internal.h>
+
+static vx_status ownCopyMoveRemoveNode(vx_graph graph, const vx_uint32 node_index, const vx_reference old_reference, const vx_reference new_reference, const vx_node bid_node);
+static vx_bool ownCopyMoveReplaceInputCompatible(vx_reference old_reference, vx_reference new_reference);
+static vx_bool ownCopyMoveReplaceOutputCompatible(vx_reference old_reference, vx_reference new_reference);
+static vx_bool ownFindRef(vx_graph graph, const vx_reference ref, const vx_enum direction, vx_node *node, vx_uint32 *index);
+static vx_status ownReassignGraphParameter(vx_graph graph, const vx_node node, const vx_uint32 index, const vx_reference ref, const vx_enum direction);
+static vx_status isUserDataCopyable(const vx_user_data_object input, const vx_user_data_object output);
```

Comment:
Need to remove these static declarations (isUserDataCopyable, copyUserData, swapUserData)

Reply:
Done